### PR TITLE
4.5 Implement get all tasks feature (for team)

### PR DESCRIPTION
**Pull request by desenzo**
Description
As a developer I want to have possibility to get all tasks for the team
and to get all tasks for the sprint.
Solution
Create service, controller and view to get all tasks for the team
and for the sprint.

### CODE REVIEW

**Comment by desenzo** [comment]
image
In team
image
in sprint 
image

File: Teams.Tests/ManageTasksServiceTest.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        public void Setup()
+        {
+            _tasksRepository = new Mock<IRepository<Task, int>>();
+            _manageTasksService = new ManageTasksService(_tasksRepository.Object);
+            var mock = getFakeDbTasks().AsQueryable().BuildMock();
+            _tasksRepository.Setup(x => x.GetAll()).Returns(mock.Object);
+        }
+
+        [Test]
+        public async System.Threading.Tasks.Task GetAllTasksInTeamAsyncReturnListTasks_ReturnListTasks()
```
Comment (SergeyMokin) [inline Teams.Tests/ManageTasksServiceTest.cs:33]: Forget in names one _ symbol, also bit weird naming
Should be: GetAllTasksInTeamAsync_RepositoryContainsTwoTasksForSelectedTeam_ReturnsTwoTasks

Also name of executed method: GetMyTaskInTeamAsync, better to use GetAllTasksForTeam

File: src/Teams/Controllers/ManageSprintsController.cs [hunk 2]
```diff
@@ -43,13 +43,14 @@ public async Task<IActionResult> AllSprints(int team_id, DisplayOptions options)
         public async Task<IActionResult> GetSprintById(int sprint_id)
         {
             var sprint = await _manageSprintsService.GetSprintAsync(sprint_id);
-            if (sprint == null)
+            if (sprint == null)
```
Comment (SergeyMokin) [inline src/Teams/Controllers/ManageSprintsController.cs:46]: please remove not needed changes

File: src/Teams/Controllers/ManageSprintsController.cs [hunk 3]
```diff
@@ -43,13 +43,14 @@ public async Task<IActionResult> AllSprints(int team_id, DisplayOptions options)
...
-            if (sprint == null)
+            if (sprint == null) 
                 return View("ErrorGetAllSprints");
             return View(sprint);
         }
         public IActionResult Index()
         {
             return View();
         }
+
```
Comment (SergeyMokin) [inline src/Teams/Controllers/ManageSprintsController.cs:54]: also here

File: Teams.Tests/ManageTasksServiceTest.cs [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        {
+            //Arrange
+            int team_id = 1;
```
Comment (SergeyMokin) [inline Teams.Tests/ManageTasksServiceTest.cs:36]: also, I missed it before, but better to use not snake case, please read doc here: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/inside-a-program/coding-conventions

should be teamId, here a lot of places to change.

snake case fine to use with consts, enum values. names for tests

Comment (SergeyMokin) [reply Teams.Tests/ManageTasksServiceTest.cs:36]: please take a look into it

File: src/Teams/Controllers/ManageTasksController.cs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            _accessCheckService = accessCheckService;
+            _manageTeamsService = manageTeamsService;
+            _manageSprintsService = manageSprintsService;
+        }
+
+        [Authorize]
+        public async Task<IActionResult> AllTasksInTeam(int team_id, DisplayOptions options)
+        {
+
+            if (await _accessCheckService.OwnerOrMemberAsync(team_id))
```
Comment (SergeyMokin) [inline src/Teams/Controllers/ManageTasksController.cs:32]: better to use less nesting.
please invert if statement, also you don't need "else" at all.
will be more readable

File: src/Teams/Services/IManageTasksService.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Teams.Models;
+
+namespace Teams.Services
+{
+    public interface IManageTasksService
+    {
+        Task<IEnumerable<Models.Task>> GetMyTaskInTeamAsync(int team_id, DisplayOptions options);
```
Comment (SergeyMokin) [inline src/Teams/Services/IManageTasksService.cs:11]: Please use different names.
Please take a look into tests comments.

File: src/Teams/Services/ManageSprintsService.cs [hunk 7]
```diff
@@ -22,7 +23,7 @@ public ManageSprintsService(IRepository<Sprint, int> sprintRepository, IManageTe
 
         public async Task<IEnumerable<Sprint>> GetAllSprintsAsync(int team_id, DisplayOptions options)
         {
-            var sprints = _sprintRepository.GetAll().Include(x => x.Team).Where(x => x.TeamId == team_id);
+            var sprints = _sprintRepository.GetAll().Include(x => x.Team).Include(x=>x.Tasks).Where(x => x.TeamId == team_id);
```
Comment (SergeyMokin) [inline src/Teams/Services/ManageSprintsService.cs:26]: Do you always need tasts included?
Maybe better to propose here bool option to include it or not?

File: src/Teams/Services/ManageSprintsService.cs [hunk 8]
```diff
@@ -36,6 +37,17 @@ public async Task<Team> GetTeam(int team_id)
             return team;
         }
 
-        public async Task<Sprint> GetSprintAsync(int sprint_id) => await _sprintRepository.GetByIdAsync(sprint_id);
+        public async Task<Sprint> GetSprintAsyncw(int sprint_id)
```
Comment (SergeyMokin) [inline src/Teams/Services/ManageSprintsService.cs:40]: typo Asyncw

File: src/Teams/Views/ManageSprints/GetSprintById.cshtml [hunk 9]
```diff
@@ -37,13 +37,22 @@
 </table>
 <table class="table">
 
+    
     <tr align="center">
-        <td align="left" style="border-top: hidden; font-weight: bold">Name of task</td>
-        <td align="left" style="border-top: hidden; font-weight: bold">Number of Story Point</td>
+        <td align="left" style="border-top: hidden; font-weight: bold">Name of Task</td>
+        <td align="left" style="border-top: hidden; font-weight: bold">User</td>
+        <td align="left" style="border-top: hidden; font-weight: bold">Store Point</td>
```
Comment (SergeyMokin) [inline src/Teams/Views/ManageSprints/GetSprintById.cshtml:44]: typo: should be Story

**Review by SergeyMokin — CHANGES_REQUESTED** [review]
Please correct the comments

File: src/Teams/Controllers/ManageTasksController.cs [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+            {
+                var tasks = await _manageTasksService.GetMyTaskInTeamAsync(team_id, options);
+                var team = await _manageTeamsService.GetTeamAsync(team_id);
+                ViewBag.TeamName = team.TeamName;
```
Comment (alehkahnovich) [inline src/Teams/Controllers/ManageTasksController.cs:36]: Please avoid using ViewBag, use model instead

**Review by alehkahnovich — CHANGES_REQUESTED** [review]

**Comment by SergeyMokin** [comment]
conflicts

File: src/Teams/Controllers/ManageSprintsController.cs [hunk 11]
```diff
@@ -42,11 +42,12 @@ public async Task<IActionResult> AllSprints(int team_id, DisplayOptions options)
         [Authorize]
         public async Task<IActionResult> GetSprintById(int sprint_id)
         {
-            var sprint = await _manageSprintsService.GetSprintAsync(sprint_id);
+            var sprint = await _manageSprintsService.GetSprintAsync(sprint_id,true);
```
Comment (SergeyMokin) [inline src/Teams/Controllers/ManageSprintsController.cs:45]: please change snake case where you working to camel

File: src/Teams/Controllers/ManageTasksController.cs [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+            IManageTeamsService manageTeamsService, IManageSprintsService manageSprintsService)
+        {
+            _manageTasksService = manageTasksService;
+            _accessCheckService = accessCheckService;
+            _manageTeamsService = manageTeamsService;
+            _manageSprintsService = manageSprintsService;
+        }
+
+        [Authorize]
+        public async Task<IActionResult> AllTasksForTeam(int team_id, DisplayOptions options)
```
Comment (SergeyMokin) [inline src/Teams/Controllers/ManageTasksController.cs:29]: team_id

File: src/Teams/Services/IManageSprintsService.cs [hunk 13]
```diff
@@ -10,6 +10,6 @@ public interface IManageSprintsService
     {
         Task<IEnumerable<Sprint>> GetAllSprintsAsync(int team_id, DisplayOptions options);
         Task<Team> GetTeam(int team_id);
-        Task<Sprint> GetSprintAsync(int sprint_id);
+        Task<Sprint> GetSprintAsync(int sprint_id, bool includeTaskAndTeamMember);
```
Comment (SergeyMokin) [inline src/Teams/Services/IManageSprintsService.cs:13]: sprint_id

File: src/Teams/Services/IManageTasksService.cs [hunk 14]
```diff
@@ -0,0 +1,13 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Teams.Models;
+
+namespace Teams.Services
+{
+    public interface IManageTasksService
+    {
+        Task<IEnumerable<Models.Task>> GetAllTasksForTeamAsync(int team_id, DisplayOptions options);
```
Comment (SergeyMokin) [inline src/Teams/Services/IManageTasksService.cs:11]: team_id

File: src/Teams/Services/ManageSprintsService.cs [hunk 15]
```diff
@@ -36,6 +37,16 @@ public async Task<Team> GetTeam(int team_id)
             return team;
         }
 
-        public async Task<Sprint> GetSprintAsync(int sprint_id) => await _sprintRepository.GetByIdAsync(sprint_id);
+        public async Task<Sprint> GetSprintAsync(int sprint_id, bool includeTaskAndTeamMember)
```
Comment (SergeyMokin) [inline src/Teams/Services/ManageSprintsService.cs:40]: sprint_id

File: src/Teams/Views/ManageSprints/GetSprintById.cshtml [hunk 16]
```diff
@@ -36,14 +36,22 @@
...
 </table>
 <table class="table">
-
+    
     <tr align="center">
-        <td align="left" style="border-top: hidden; font-weight: bold">Name of task</td>
-        <td align="left" style="border-top: hidden; font-weight: bold">Number of Story Point</td>
+        <td align="left" style="border-top: hidden; font-weight: bold">Name of Task</td>
+        <td align="left" style="border-top: hidden; font-weight: bold">User</td>
+        <td align="left" style="border-top: hidden; font-weight: bold">Store Point</td>
```
Comment (SergeyMokin) [inline src/Teams/Views/ManageSprints/GetSprintById.cshtml:43]: Store Point

File: src/Teams/Views/ManageTasks/AllTasksForTeam.cshtml [hunk 17]
```diff
@@ -0,0 +1,32 @@
...
+</div>
+<div align="center">
+    <h><font size="6" color="black"><strong>@Model.First().Team.TeamName</strong></font></h>
+</div>
+<table class="table">
+
+    <tr align="center">
+        <td align="left" style="border-top: hidden; font-weight: bold">Name of Task</td>
+        <td align="left" style="border-top: hidden; font-weight: bold">User</td>
+        <td align="left" style="border-top: hidden; font-weight: bold">Store Point</td>
```
Comment (SergeyMokin) [inline src/Teams/Views/ManageTasks/AllTasksForTeam.cshtml:19]: store point

**Comment by SergeyMokin** [comment]
please fix conflicts

**Review by SergeyMokin — APPROVED** [review]

**Pull request merged**